Framework/Algorithms/test/ConvertTableToMatrixWorkspaceTest.h

```diff
@@ -139,6 +139,88 @@ class ConvertTableToMatrixWorkspaceTest : public CxxTest::TestSuite {
     TS_ASSERT_THROWS_NOTHING(m_converter->setPropertyValue("ColumnX", "A"));
     TS_ASSERT_THROWS_NOTHING(m_converter->setPropertyValue("ColumnY", "B"));
   }
+  // test if it can convert a string to a double, if the string is numeric
+  void testStringToDouble() {
+
+    ITableWorkspace_sptr tws = WorkspaceFactory::Instance().createTable();
+    tws->addColumn("str", "A");
+    tws->addColumn("double", "B");
+    tws->addColumn("double", "C");
+
+    size_t n = 10;
+    for (size_t i = 0; i < n; ++i) {
+      TableRow row = tws->appendRow();
+      std::string x = "1";
+      double y = double(i) * 1.1;
```

Comment:
In general we avoid functional-style casts (and especially c-style casts) and use the new-style casting. Clearly here it won't make a huge difference, but you could consider it consistent with the rest of Mantid.